castai/client.go

```diff
@@ -23,11 +24,19 @@ const (
 type Client interface {
 	GetActions(ctx context.Context, k8sVersion string) ([]*ClusterAction, error)
 	AckAction(ctx context.Context, actionID string, req *AckClusterActionRequest) error
-	SendLogs(ctx context.Context, req *LogEvent) error
 	SendAKSInitData(ctx context.Context, req *AKSInitDataRequest) error
 }
 
-func NewClient(log *logrus.Logger, rest *resty.Client, clusterID string) Client {
+type client struct {
+	log       *logrus.Logger
+	rest      *resty.Client
+	clusterID string
+}
+
+// client must satisfy logexporter.Sender.
+var _ logexporter.Sender = new(client)
+
+func NewClient(log *logrus.Logger, rest *resty.Client, clusterID string) *client {
```

Comment:
Public func returns a public struct
make client public

Reply:
type itself yes, but fields are still private, aren't they?

Reply:
Yep I will make the type public and fields will stay private.